Framework/Algorithms/src/SampleCorrections/MayersSampleCorrection.cpp

```diff
@@ -134,9 +131,10 @@ void MayersSampleCorrection::exec() {
     params.sigmaSc = sampleMaterial.totalScatterXSection();
     params.cylRadius = radius;
     params.cylHeight = height;
-    MayersSampleCorrectionStrategy correction(params, inX, inputWS->readY(i),
-                                              inputWS->readE(i));
-    correction.apply(outputWS->dataY(i), outputWS->dataE(i));
+
+    MayersSampleCorrectionStrategy correction(params, inputWS->histogram(i));
+    outputWS->setHistogram(i,
+                           correction.apply(std::move(inputWS->histogram(i))));
```

Comment:
I looked at moving the entire class into a function but the class has several private methods which do the calculations for it. 
Looking at the usage stats the algorithm has been used around 10 times since 3.5 so I'm reluctant to completely tidy it however the changes (I'm yet to push) make it somewhat better.